app/domain/authentication/authn_k8s/k8s_resolver.rb

```diff
@@ -74,92 +65,94 @@ def validate_pod
       # Tests whether the Pod's ReplicaSet belongs to the Deployment.
       class Deployment < Base
         def validate_pod
-          replica_set_ref = verify "Pod #{pod_name} does not belong to a ReplicaSet (or Deployment)" do
-            pod_owner_refs &&
-              pod_owner_refs.find{|ref| ref.kind == "ReplicaSet"}
+          replica_set_ref = pod_owner_refs.find{|ref| ref.kind == "ReplicaSet"}
+          unless pod_owner_refs && replica_set_ref
+            raise Err::PodMissingRelationError.new(pod_name, 'ReplicaSet')
           end
-          
+
           replica_set = k8s_object_lookup.find_object_by_name "replica_set", replica_set_ref.name, namespace
 
-          deployment_ref = verify "Pod #{pod_name} does not belong to a Deployment" do
-            replica_set.metadata.ownerReferences &&
-              replica_set.metadata.ownerReferences.find{|ref| ref.kind == "Deployment"}
+          deployment_ref = replica_set.metadata.ownerReferences.find{|ref| ref.kind == "Deployment"}
```

Comment:
this can raise a nil error if `replica_set.metadata.ownerReferences` is nil. please maintain the logic

Reply:
you can change to:
```
unless replica_set_owner_refs && deployment_def
  raise Err::PodMissingRelationError.new(pod_name, 'Deployment')
end

def deployment_def
  @deployment_def ||= replica_set_owner_refs.find{|ref| ref.kind == "Deployment"}
end

def replica_set_owner_refs
  @replica_set_owner_refs ||= replica_set.metadata.ownerReferences
end
```

Reply:
replica_set_owner_refs should accept replica_set as arg